Wikipedia/Code/WMFAppViewController.m

```diff
@@ -1215,9 +1215,21 @@ - (BOOL)processUserActivity:(NSUserActivity *)activity animated:(BOOL)animated c
             [self.currentTabNavigationController popToRootViewControllerAnimated:animated];
             NSURL *articleURL = activity.wmf_linkURL;
             if (articleURL) {
-                // For "View on a map" action to succeed, view mode has to be set to map.
-                [[self placesViewController] updateViewModeToMap];
-                [[self placesViewController] showArticleURL:articleURL];
+                double latitude = [[articleURL wmf_valueForQueryKey:@"latitude"] doubleValue];
+                double longitude = [[articleURL wmf_valueForQueryKey:@"longitude"] doubleValue];
+                NSString *place = [articleURL wmf_valueForQueryKey:@"place"];
+
+                if (latitude && longitude) {
+                    // If the place coordinate is provided in the URL, show it in the map
+                    [[self placesViewController] searchPlaceWithCoordinateWithLatitude:latitude longitude:longitude];
```

Comment:
I am not able to hit this condition - I think the latitude and longitude values are getting dropped somewhere.